### PR TITLE
pacman.cpp: Remove diplocations from edge connectors/split-pads in pacman, add dip locations to mspacman, remove duplicated inputs from mspacpls and mspachmp, add publisher to titanpac

### PR DESCRIPTION
**Pull request by einstein95**
Interestingly, it appears that NSM was a licensed distributor of Pac-Man in West Germany at the time according to a notice from Namco in Game Machine

### CODE REVIEW

File: src/mame/drivers/pacman.cpp [hunk 1]
```diff
@@ -1595,11 +1595,11 @@ static INPUT_PORTS_START( mspacman )
...
 	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_4WAY
 	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_4WAY
-	PORT_DIPNAME( 0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1)
+	PORT_DIPNAME( 0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1) PORT_DIPLOCATION("SW:7")
 	PORT_DIPSETTING(    0x10, DEF_STR( Off ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
-	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_COIN1 )
-	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_COIN2 )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_COIN1 ) /* Also invincibility in mspacpls */
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_COIN2 ) /* Also speed-up in mspacpls */
```
Comment (0kmg) [inline src/mame/drivers/pacman.cpp:1602]: Coin inputs give invincibility and speed-up? Was this a copy/paste meant to be on INP_START1/2?

Also may as well make these C++ style comments while we're moving them :wink:

Comment (DavidHaywood) [reply src/mame/drivers/pacman.cpp:1602]: makes me wonder if these were multiplexed somehow, or if they're really just debug features.  A moment of speedup and/or invincibility when inserting a coin seems like an odd choice.

Comment (einstein95) [reply src/mame/drivers/pacman.cpp]: Ah yeah, was a silly copy/paste mistake. Fixed and converted the comments

File: src/mame/drivers/pacman.cpp [hunk 2]
```diff
@@ -1575,10 +1575,10 @@ static INPUT_PORTS_START( pacman )
...
 	PORT_DIPSETTING(    0x20, "20000" )
 	PORT_DIPSETTING(    0x30, DEF_STR( None ) )
-	PORT_DIPNAME( 0x40, 0x40, DEF_STR( Difficulty ) )   PORT_DIPLOCATION("SW:7") // physical location for difficulty on puckman set is split-pad between R32 and C29
+	PORT_DIPNAME( 0x40, 0x40, DEF_STR( Difficulty ) )   // physical location for difficulty on puckman set is split-pad between R32 and C29
 	PORT_DIPSETTING(    0x40, DEF_STR( Normal ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( Hard ) )
-	PORT_DIPNAME( 0x80, 0x80, "Ghost Names" )           PORT_DIPLOCATION("SW:8") // physical location for ghostnames on puckman set is split-pad between C10 and C29
+	PORT_DIPNAME( 0x80, 0x80, "Ghost Names" )           // physical location for ghostnames on puckman set is split-pad between C10 and C29
 	PORT_DIPSETTING(    0x80, DEF_STR( Normal ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( Alternate ) )
```
Comment (cuavas) [inline src/mame/drivers/pacman.cpp:1583]: Please make these either machine configuration or toggle inputs if they aren’t actually DIP switches.  MAME has a menu for toggle inputs now, so they aren’t completely opaque any more.

File: src/mame/drivers/pacman.cpp [hunk 3]
```diff
@@ -1595,7 +1595,7 @@ static INPUT_PORTS_START( mspacman )
 	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_4WAY
 	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_4WAY
 	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_4WAY
-	PORT_DIPNAME( 0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1)
+	PORT_DIPNAME( 0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1) PORT_DIPLOCATION("SW:7")
 	PORT_DIPSETTING(    0x10, DEF_STR( Off ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
```
Comment (cuavas) [inline src/mame/drivers/pacman.cpp:1600]: Please make this a toggle input, too.

File: src/mame/drivers/pacman.cpp [hunk 4]
```diff
@@ -1895,7 +1798,7 @@ static INPUT_PORTS_START( mspactwin )
 	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_4WAY PORT_PLAYER(2)
 	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_4WAY PORT_PLAYER(2)
 	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_4WAY PORT_PLAYER(2)
-	PORT_DIPNAME( 0x80, 0x80, "Skip Screen" ) // Used to skip level
+	PORT_DIPNAME( 0x80, 0x80, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1)
 	PORT_DIPSETTING(    0x80, DEF_STR( Off ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
```
Comment (cuavas) [inline src/mame/drivers/pacman.cpp:1803]: Please make this a toggle input.

**Review by cuavas — CHANGES_REQUESTED** [review]
Please don’t use DIP switches for things that aren’t DIP switches.  You can use machine configuration inputs or toggle inputs.  Toggle inputs now have a menu so they’re more accessible than in previous versions of MAME.

File: src/mame/drivers/pacman.cpp (hunk 2 above)
Comment (einstein95) [reply src/mame/drivers/pacman.cpp]: Changed to machine configs

File: src/mame/drivers/pacman.cpp (hunk 3 above)
Comment (einstein95) [reply src/mame/drivers/pacman.cpp]: Done

File: src/mame/drivers/pacman.cpp (hunk 4 above)
Comment (einstein95) [reply src/mame/drivers/pacman.cpp]: Done

File: src/mame/drivers/pacman.cpp [hunk 5]
```diff
@@ -1540,7 +1540,7 @@ static INPUT_PORTS_START( pacman )
 	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT )  PORT_PLAYER(1) PORT_4WAY
 	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_PLAYER(1) PORT_4WAY
 	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN )  PORT_PLAYER(1) PORT_4WAY
-	PORT_DIPNAME(0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1)
+	PORT_DIPNAME(0x10, 0x10, "Rack Test (Cheat)" ) PORT_CODE(KEYCODE_F1) PORT_DIPLOCATION("SW:7")
 	PORT_DIPSETTING(   0x10, DEF_STR( Off ) )
 	PORT_DIPSETTING(   0x00, DEF_STR( On ) )
```
[the remark this answers is not in the document]
Comment (einstein95) [reply src/mame/drivers/pacman.cpp]: Done

**Pull request merged**